wordpress-mobile/WordPress-Android
# [Blogging Reminders Sync] Implement blogging reminders resolver

Fixes #17279

To test:
1 - Install WordPress and log in;
2 - Add blogging prompts to a couple of different sites;
3 - Set the flag `isFeatureFlagEnabled` to `true` on `SharedLoginResolver.kt`, `UserFlagsResolver.kt` **and** `BloggingRemindersResolver.kt` (this cannot be done using the `Debug settings` screen because that is only available when the user is already logged in);
4 - Replace `jetpackProviderSyncFeatureConfig.isEnabled()` with `true` on `SharedLoginProvider.kt`, `UserFlagsProvider.kt` **and** `BloggingRemindersProvider.kt`.
5 - Install Jetpack using the same variant (e.g. if installed WordPress version was `jalapenoDebug`, install Jetpack `jalapenoDebug`);
6 - Jetpack should login automatically after launching, and home screen should be shown;
7 - Check if the blogging prompts set on step 2 were set correctly in the site settings.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit tests.

3. What automated tests I added (or what prevented me from doing so)
Implemented/updated: `BloggingRemindersSyncAnalyticsTrackerTest.kt`, `JetpackBloggingRemindersSyncFlagTest.kt`, `BloggingRemindersResolverTest.kt`, `QueryResultTest.kt` and `ClientVerificationTest.kt`.

PR submission checklist:
